src/Extensions/Microsoft.Diagnostics.Monitoring.Extension.Common/EgressHelper.cs

```diff
@@ -35,8 +37,24 @@ private static async Task<int> Egress<TOptions>(EgressProvider<TOptions> provide
             EgressArtifactResult result = new();
             try
             {
-                string jsonConfig = Console.ReadLine();
-                ExtensionEgressPayload configPayload = JsonSerializer.Deserialize<ExtensionEgressPayload>(jsonConfig);
+                StdInStream = Console.OpenStandardInput();
+
+                var payloadVersionBuffer = stackalloc byte[sizeof(int)].ToArray();
```

Comment:
```suggestion
                Span<byte> payloadVersionSpan = stackalloc byte[sizeof(int)];
```

Using `ToArray` defeats the purpose of using `stackalloc`.

Reply:
That's what I tried initially, but I got "Parameters or locals of type 'Span<byte>' cannot be declared in async methods or async lambda expressions."

Reply:
Totally forgot this is in an async method; arrays allocated with `stackalloc` cannot be boxed since they are purely on the stack and the async state machine hoists its locals to properties. Just allocate the arrays without using `stackalloc`.